_posts/2021-11-22-using-context-to-simplify-a-react-form.md

```diff
@@ -0,0 +1,132 @@
+   exactly the props needed in the component (if there were any.)
+
+   1. A note on tests: Whenever we took advantage of the Formik context in a component, we were breaking that
+      component's tests, because the wrappers we were using in the tests did not have access to the Formik context
+      and were being passed props that the component was no longer looking at. We created a helper test wrapper
+      that we could use in all of our Artwork Form tests to wrap the test's specific wrapper inside of a
+      `<Formik/>` component and provide the specific `values` to use as the initial values for `Formik`. Many of
+      our test cases involve passing different `values` into the component, which we originally did via props but
+      now will do via the Formik context through the wrapper. Here's what the wrapper looks like:
+
+   ```tsx
+   export const TestFormikWrapper: React.FC<TestFormikWrapperProps> = (props) => {
+     const { children, values } = props
+
+     return (
+       <Formik initialValues={values} onSubmit={jest.fn()}>
+         {children}
+       </Formik>
+     )
+   }
+   ```
+
+4. Once we completed the conversion all the way up the tree to the root component, the `ArtworkForm`, we typed that
+   component as strictly as possible and made sure to get rid of `any`'s. There were quite a few when we started
+   the process.
+
+So, where did we end up? Now all of the components in the Artwork Form are making use of the `useArtworkForm` hook
+if they were previously accessing any of the values from the Formik context from props. We have much less prop
+drilling between components and instead explicitly pass down the props needed from parent to child. It's now much
+more clear for developers what data is passing between the components and what data is actually being used in the
+child. All of the components are also explicitly typed so we know exactly which props, if any, need to be passed
+down from the parent. If any of these props are removed, TypeScript helps us by failing loudly.
+
+One of the main pain points of the Artwork Form is that it's very difficult for new developers (whether new to
+Artsy or new to the Partner Experience team) to contribute and make changes to the form without breaking something
+or spending extra time figuring out how data is passed within the form. Hopefully, this change will make it easier
+for developers to understand the Artwork From.
+
+How did the Artwork Form get so complicated? Well, as we shared, the Artwork Form is the key to achieving one of
+the PX team's core goals: surfacing the most accurate and rich information about artworks to collectors. We have to
+allow partners to add more and increasingly specific pieces of metadata to artworks. The form has been growing and
+for better or worse, will need to keep growing. Hopefully, this refactor will allow us to expand the form more
```

Comment:
Is there anything to be said in here for the fact that volt doesn't get as much design/ux attention as our collector-facing apps?